<launch>

    <!-- You should edit these parameters to point to your topics -->

    <arg name="camera_topic" default="/cam_repub" doc="Camera topic name" />
    <arg name="cmd_vel_pub" default="/bluetooth_teleop/cmd_vel" doc="The topic that cmd_vel messages will be published on when replaying a map (can be the same as cmd_vel_sub)" />
    <arg name="cmd_vel_sub" default="/bluetooth_teleop/cmd_vel" doc="The topic that cmd_vel messages will be recorded from when making a map (can be the same as cmd_vel_pub)" />
    <arg name="odom_topic" default="/robot1/odometry" doc="The topic that odometry information will be read from" />
    <arg name="additional_record_topics" default="/robot1/odometry" doc="Use this parameter to record any additional topics you wish to be in your map and repeated each traversal. Space separated topics names." />

    <!-- You should not need to edit anything below this line. -->

    <arg name="particle_num" default="600" doc="Number of particles" />
    <arg name="odom_error" default="0.05" doc="Percentual odometry inaccuracy" />
    <arg name="dist_init_std" default="1.0" doc="Standart deviation for particle initial spawn in distance dimension" />
    <arg name="align_beta" default="30.0" doc="Scaling factor of individual histograms" />
    <arg name="align_init_std" default="0.3" doc="Standart deviation for particle initial spawn in displacement dimension" />
    <arg name="add_random" default="0.1" doc="Percentage of randomly (uniformly) spawned particles each timestep" />
   
    <!-- Tunable parameters -->

    <arg name="choice_beta" default="3.0" doc="Scaling factor for likelihoods of individual particles - higher value makes filtering to choose only particles with high likelihoods. Effectively, higher the value the more is used visual feedback and less odometry. We suggest 1.5-2.0 for large outdoor environments and 3.0 for indoor." />
    <arg name="matching_type" default="siam" doc="representation matching scheme - use siam or siam_f" />

	<group ns="pfvtr">

        <node pkg="pfvtr" type="representations-ros-1.py" name="representations" respawn="true" output="screen" args="-OO" >
            <param name="camera_topic" type="string" value="$(arg camera_topic)" />
            <param name="matching_type" type="string" value="$(arg matching_type)" />
        </node>

    </group>
</launch>
